net-test/bin/start.sh

```diff
@@ -0,0 +1,363 @@
+   fi
+
+   logln "PID $BITCOIN_NEON_CONTROLLER_PID"
+
+   echo "$BITCOIN_NEON_CONTROLLER_PID"
+   return 0
+}
+
+start_faucet() {
+   logln "Setting up Bitcoin faucet..."
+   test -f "$BITCOIN_CONTROLLER_CONF" || exit_error "No such file or directory: $BITCOIN_CONTROLLER_CONF"
+   test -f "$BITCOIN_CONF" || exit_error "No such file or directory: $BITCOIN_CONF"
+
+   local PRIVKEY="$(cat "$BITCOIN_CONTROLLER_CONF" | grep "# WIF" | cut -d ' ' -f 3)"
+
+   log "  Importing faucet private key..."
+   bitcoin-cli -conf="$BITCOIN_CONF" importprivkey "$PRIVKEY" >"$FAUCET_LOGFILE" 2>&1
+   local RC=$?
+   if [ $RC -ne 0 ]; then
+      logln "FAILED!"
+      return 1
+   fi
+   logln "ok"
+
+   local MINER_ADDR="$(cat "$BITCOIN_CONTROLLER_CONF" | grep "miner_address =" | cut -d ' ' -f 3 | sed -r 's/"//g')"
+
+   log "  Importing miner address $MINER_ADDR..."
+   
+   echo "Try importing miner address $MINER_ADDR" >>"$FAUCET_LOGFILE"
+   bitcoin-cli -conf="$BITCOIN_CONF" importaddress "$MINER_ADDR" >>"$FAUCET_LOGFILE" 2>&1
+   RC=$?
+   if [ $RC -ne 0 ]; then 
+      logln "FAILED!"
+      return 1
+   fi
+
+   logln "ok"
+
+   log "Starting Bitcoin faucet..."
+   faucet.sh "$FAUCET_PORT" "$BITCOIN_CONF" >"$FAUCET_LOGFILE" 2>&1 &
+   local FAUCET_PID=$!
+   logln "PID $FAUCET_PID"
+
+   echo "$FAUCET_PID"
+   return 0
+}
+
+start_stacks_master_node() {
+   logln "Setting up Stacks master node..."
+   test -f "$STACKS_MASTER_CONF".in || exit_error "No such file or directory: $STACKS_MASTER_CONF.in"
+
+   log "  Generating Stacks master config file..."
+   sed -r \
+      -e "s!@@STACKS_CHAINSTATE_DIR@@!$STACKS_CHAINSTATE_DIR!g" \
+      -e "s!@@STACKS_PUBLIC_IP@@!$STACKS_MASTER_PUBLIC_IP!g" \
+      "$STACKS_MASTER_CONF".in \
+      > "$STACKS_MASTER_CONF"
+   logln "ok"
+
+   log "Starting Stacks master node..."
+   BLOCKSTACK_DEBUG=1 RUST_BACKTRACE=full stacks-node start --config="$STACKS_MASTER_CONF" >"$STACKS_MASTER_LOGFILE" 2>&1 &
+   local STACKS_PID=$!
+   logln "PID $STACKS_PID"
+
+   echo "$STACKS_PID"
+   return 0
+}
+
+start_stacks_miner_node() {
+   logln "Setting up Stacks miner node..."
+   test -f "$STACKS_MINER_CONF".in || exit_error "No such file or directory: $STACKS_MINER_CONF.in"
+
+   log "  Generating Stacks miner config file..."
+   local STACKS_MASTER_IP="$1"
+   local BITCOIN_IP="$2"
+   local FAUCET_URL="$3"
+   local STACKS_MINER_SEED="$(blockstack-cli generate-sk | jq -r '.secretKey')"
+
+   sed -r \
+      -e "s/@@STACKS_MASTER_IP@@/$STACKS_MASTER_IP/g" \
+      -e "s/@@BITCOIN_IP@@/$BITCOIN_IP/g" \
+      -e "s/@@STACKS_MINER_SEED@@/$STACKS_MINER_SEED/g" \
+      -e "s!@@STACKS_CHAINSTATE_DIR@@!$STACKS_CHAINSTATE_DIR!g" \
+      "$STACKS_MINER_CONF".in \
+      > "$STACKS_MINER_CONF"
+   logln "ok"
+
+   local BTCADDR="$(blockstack-cli --testnet addresses "$STACKS_MINER_SEED" | jq -r '.BTC')"
+   local TIMEOUT=2
+   local TXID=""
+   local CONFIRMATIONS=0
+
+   log "  Fetching BTC from the faucet for this miner ($BTCADDR)..."
```

Comment:
before I got the faucet to work, I replace the code with
```
  bitcoin-cli --regtest -conf="$BITCOIN_CONF" sendtoaddress $BTCADDR 2
  logln "waiting 30s"
  sleep 30
```
and this removes the need to have the faucet, even though this is quite an achievement

in theory, in regtest mode, you should be able to use `generatetoaddress`, but somehow I could not get it to work.

Reply:
The `faucet.sh` program will ultimately be used to service hundreds (if not thousands) of miners booting up and fetching BTC, and each miner will have its own unique private key that it generates itself (per `start_stacks_miner_node()`).  So, `faucet.sh` is here to stay.  Also, I'm probably going to expand it (and possibly rename it) so it can handle various queries on the state of the running node that currently aren't handled by the RESTful API.